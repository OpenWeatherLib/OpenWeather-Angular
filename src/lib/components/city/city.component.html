<mat-card class="ga-card" *ngIf="city">
  <mat-card-header>
    <img mat-card-avatar class="ga-card-header-image" src="{{cityPictureUrl}}" alt="City pic">
    <mat-card-title>City</mat-card-title>
    <mat-card-subtitle>{{city?.name}}</mat-card-subtitle>
  </mat-card-header>
  <img mat-card-image src="{{cityPictureUrl}}" alt="City pic">
  <mat-card-content>
    <p>Country: {{city?.country}}</p>
    <p>Population: {{city?.population}}</p>
    <p>Coordinates: Lat: {{city?.coord?.lat}}, Lon: {{city?.coord?.lon}}</p>
  </mat-card-content>
</mat-card>
