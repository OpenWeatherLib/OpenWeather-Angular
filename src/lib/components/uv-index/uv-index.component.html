<mat-card class="ga-card" *ngIf="uvIndex">
  <mat-card-header>
    <img mat-card-avatar class="ga-card-header-image" src="/assets/uv_index.png" alt="UvIndex pic">
    <mat-card-title>UvIndex</mat-card-title>
    <mat-card-subtitle>{{uvIndex?.value}}</mat-card-subtitle>
  </mat-card-header>
  <mat-card-content>
    <p>Coordinates: Lat: {{uvIndex?.lat}}, Lon: {{uvIndex?.lon}}</p>
  </mat-card-content>
</mat-card>
