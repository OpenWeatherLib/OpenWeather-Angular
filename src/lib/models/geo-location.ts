import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class GeoLocation {
  lat: number = 720.0;
  long: number = 720.0;

  isDefault(): boolean {
    return this.lat === 720.0 && this.long === 720.0;
  }
}
