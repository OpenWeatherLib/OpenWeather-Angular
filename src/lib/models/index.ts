export * from "./address-component";
export * from "./city";
export * from "./city2";
export * from "./clouds";
export * from "./coordinates";
export * from "./coordinates2";
export * from "./geometry";
export * from "./main";
export * from "./rain";
export * from "./sys";
export * from "./weather-current";
export * from "./weather-forecast";
export * from "./weather-forecast-part";
export * from "./weather-part";
export * from "./wind";
export * from "./uv-index";
