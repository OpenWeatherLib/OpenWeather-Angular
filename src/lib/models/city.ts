import { GeoLocation } from "./geo-location";
import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class City {
  id: number = 0;
  name: string = "";
  country: string = "";
  population: number = 0;
  geoLocation: GeoLocation = new GeoLocation();

  isDefault(): boolean {
    return this.id === 0 && this.name === "" && this.country === "" && this.population === 0 && this.geoLocation.isDefault();
  }
}
