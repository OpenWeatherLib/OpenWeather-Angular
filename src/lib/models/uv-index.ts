import { GeoLocation } from "./geo-location";

export class UvIndex {
  geoLocation: GeoLocation = new GeoLocation();
  date: Date = new Date();
  value: number = 0.0;

  isDefault(): boolean {
    return this.geoLocation.isDefault() && this.value === 0.0;
  }
}
