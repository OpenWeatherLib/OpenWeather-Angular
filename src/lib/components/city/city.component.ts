import { Component, Input, OnInit } from "@angular/core";

import { BaseComponent } from "@lib/components/base-component/base.component";
import { City } from "@lib/models";
import { ImageService, OpenWeatherService } from "@lib/services";

@Component({
  selector: "ga-city",
  templateUrl: "./city.component.html",
  styleUrls: ["./city.component.scss"]
})
export class CityComponent extends BaseComponent implements OnInit {

  @Input() initialCityName: string = String().empty;

  city: City = null;
  cityPictureUrl: string = String().empty;

  constructor(private readonly imageService: ImageService,
    private readonly openWeatherService: OpenWeatherService) {
    super();
  }

  ngOnInit() {
    this.openWeatherService.loadCityData(this.initialCityName);

    this.registerSubscription(
      this.openWeatherService.city()
        .subscribe(city => {
          if (city) {
            this.city = city;
            this.imageService.receiveImagePictureUrl(this.city.name);
          }
        }));

    this.registerSubscription(
      this.imageService.cityPictureUrl()
        .subscribe(url => {
          this.cityPictureUrl = url;
        }));
  }
}
