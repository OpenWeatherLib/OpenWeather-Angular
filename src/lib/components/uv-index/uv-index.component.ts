import { Component, OnInit } from "@angular/core";
import { Store } from "@ngrx/store";
import { Observable } from "rxjs";

import { UvIndex } from "@lib/models";
import { selectIsLoading, selectUvIndex } from "@lib/store/uv-index-store";
import { RootState } from "@lib/store/root-state";

@Component({
  selector: "ga-uv-index",
  templateUrl: "./uv-index.component.html",
  styleUrls: ["./uv-index.component.scss"]
})
export class UvIndexComponent implements OnInit {

  isLoading$: Observable<boolean>;

  uvIndex$: Observable<UvIndex>;

  constructor(private readonly store$: Store<RootState>) { }

  ngOnInit() {
    this.isLoading$ = this.store$.select(selectIsLoading);

    this.uvIndex$ = this.store$.select(selectUvIndex);
  }
}
