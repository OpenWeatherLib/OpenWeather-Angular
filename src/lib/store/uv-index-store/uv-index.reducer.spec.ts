import { City, UvIndex } from "@lib/models";
import { loadCitySuccessAction } from "../city-store/city.actions";
import { loadUvIndexRequestAction, loadUvIndexSuccessAction, loadUvIndexErrorAction } from "./uv-index.actions";
import { uvIndexReducer } from "./uv-index.reducer";
import { UvIndexState } from "./uv-index.state";

describe("UvIndex Reducer Tests", () => {

    const initialState: UvIndexState = {
        uvIndex: undefined,
        isLoading: false,
        error: "Test Error"
    };

    describe("State Changes", () => {
        test("should have initial state after invalid action", () => {
            // Arrange & Act
            const testState: UvIndexState = uvIndexReducer(initialState, { type: "INVALID_ACTION" } as any);

            // Assert
            expect(testState).toBe(initialState);
        });

        test("should load city on success of loading", () => {
            // Arrange
            const loadRequest = loadUvIndexRequestAction({ city: {} as City });
            const loadSuccess = loadUvIndexSuccessAction({ uvIndex: {} as UvIndex });

            // Act
            const testState: UvIndexState = [loadRequest, loadSuccess].reduce(uvIndexReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set error on error of loading", () => {
            // Arrange
            const loadRequest = loadUvIndexRequestAction({ city: {} as City });
            const loadError = loadUvIndexErrorAction({ error: "Error" });

            // Act
            const testState: UvIndexState = [loadRequest, loadError].reduce(uvIndexReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set loading to true on loadCitySuccessAction", () => {
            // Arrange
            const loadSuccess = loadCitySuccessAction({ city: {} as City });

            // Act
            const testState: UvIndexState = [loadSuccess].reduce(uvIndexReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });
    });
});
