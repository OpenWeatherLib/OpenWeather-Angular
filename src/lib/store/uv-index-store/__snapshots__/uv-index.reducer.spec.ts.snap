// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UvIndex Reducer Tests State Changes should load city on success of loading 1`] = `
Object {
  "error": undefined,
  "isLoading": false,
  "uvIndex": Object {},
}
`;

exports[`UvIndex Reducer Tests State Changes should set error on error of loading 1`] = `
Object {
  "error": "Error",
  "isLoading": false,
  "uvIndex": undefined,
}
`;

exports[`UvIndex Reducer Tests State Changes should set loading to true on loadCitySuccessAction 1`] = `
Object {
  "error": "Test Error",
  "isLoading": true,
  "uvIndex": undefined,
}
`;
